src/components/CommentList.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { ErrorType } from '../types/ErrorType';
+import { customFetch } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+import { CommentInfo } from './CommentInfo';
+
+type Props = {
+  comments: Comment[];
+  setErrorType: React.Dispatch<React.SetStateAction<ErrorType | null>>;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+};
+
+export const CommentsList: React.FC<Props> = ({
+  comments,
+  setComments,
+  setErrorType,
+}) => {
+  const handleDeleteComment = async (commentId: number) => {
+    try {
+      setComments(prevComments => prevComments
```

Comment:
The deletion of a comment is optimistic. The comment is removed from the UI before the API confirms that it was successfully deleted. If the API call fails, the UI and the server state will be out of sync.